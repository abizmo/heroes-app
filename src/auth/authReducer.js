import { LOGIN, LOGOUT } from "./authActionTypes"

const authReducer = (state = {}, { type, payload }) => {
  switch (type) {
    case LOGIN:
      return { ...payload, logged: true }

    case LOGOUT:
      return { logged: false }

    default:
      return state
  }
}

export default authReducer
