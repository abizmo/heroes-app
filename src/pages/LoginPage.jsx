import React from 'react'

const LoginPage = ({ history }) => {
  const handleLogin = () => {
    history.replace("/")
  }

  return (
    <div className="login text-center">
      <main className="form-signin">
        <form>
          <h1 className="h3 mb-3 fw-normal">Please login</h1>
          <button
            className="w-100 btn btn-lg btn-primary"
            onClick={handleLogin}
            type="submit"
          >Login</button>
          <p className="mt-5 mb-3 text-muted">&copy; 2021</p>
        </form>
      </main>
    </div>
  )
}

export default LoginPage
