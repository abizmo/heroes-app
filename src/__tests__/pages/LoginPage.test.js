import { mount } from "enzyme"
import { createMemoryHistory } from 'history'
import { Router } from "react-router"
import { LOGIN } from "../../auth/authActionTypes"
import AuthContext from "../../auth/AuthContext"
import LoginPage from "../../pages/LoginPage"

Storage.prototype.getItem = jest.fn()

describe('LoginPage tests', () => {
  const dispatch = jest.fn()
  const evt = {
    preventDefault: () => {}
  }

  test('should render and match snapshot', () => {
    const wrapper = mount(
      <AuthContext.Provider value={{ dispatch }}>
        <LoginPage />
      </AuthContext.Provider>
    )

    expect(wrapper.find('button').text()).toBe('Login')
    expect(wrapper).toMatchSnapshot()
  })
  
  test('should make login, call localStorage.getItem, dispatch and history.replace', () => {
    const action = {
      type: LOGIN,
      payload: {
        username: 'Goccita'
      },
    }
    const history = createMemoryHistory()
    history.replace = jest.fn()
    const wrapper = mount(
      <AuthContext.Provider value={{ dispatch }}>
        <Router history={history}>
          <LoginPage />
        </Router>
      </AuthContext.Provider>
    )
    
    wrapper.find('form').prop('onSubmit')(evt)
    expect(dispatch).toHaveBeenCalledWith(action)
    expect(history.replace).toHaveBeenCalled()
    expect(localStorage.getItem).toHaveBeenCalledWith('lastPath')
  })
})
