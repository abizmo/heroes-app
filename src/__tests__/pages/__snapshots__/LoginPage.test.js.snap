// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage tests should render and match snapshot 1`] = `
<div
  className="login text-center"
>
  <main
    className="form-signin"
  >
    <form
      onSubmit={[Function]}
    >
      <h1
        className="h3 mb-3 fw-normal"
      >
        Please login
      </h1>
      <button
        className="w-100 btn btn-lg btn-primary"
        type="submit"
      >
        Login
      </button>
      <p
        className="mt-5 mb-3 text-muted"
      >
        Â© 2021
      </p>
    </form>
  </main>
</div>
`;
